### PR TITLE
feat: add dataset iam attach module

### PR DESCRIPTION
**Pull request by maetolay**
Please confirm that you have done the following before requesting reviews:

- [x] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [x] I have typed an adequate description that explains **why** I am making this change.
- [x] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

add module where we attache I am and policy into a created dataset

---
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-gcp-bigquery/3)

### CODE REVIEW

**Review by nithipanhn — APPROVED** [review]

File: example/create_dataset/main.tf [hunk 1]
```diff
@@ -4,23 +4,28 @@ terraform {
...
       version = "3.1.2"
     }
   }
 }
 
+resource "random_id" "random_id" {
+  byte_length = 4
+}
+
+resource "google_service_account" "google_service_account" {
```
Comment (jai) [inline example/create_dataset/main.tf:18]: this name can definitely be improved. The type is already `google_service_account` the name should be something more descriptive.

**Review by jai — CHANGES_REQUESTED** [review]
pls improve naming

File: example/attach_iam_to_dataset/main.tf [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+resource "random_id" "random_id" {
+  byte_length = 4
+}
+
+resource "google_service_account" "owner" {
+  account_id   = "datasetowner${random_id.random_id.hex}"
+  display_name = "dataset_owner"
+}
+
+module "crate_bigquery_dataset" {
```
Comment (jai) [inline example/attach_iam_to_dataset/main.tf:23]: `crate` > `create`

File: example/attach_iam_to_dataset/main.tf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  source = "../../modules/gcp_bigquery_dataset"
+
+  name        = "dataset"
+  description = "dataset's description"
```
Comment (jai) [inline example/attach_iam_to_dataset/main.tf:27]: what's with this hardcoding of...nothing?

File: example/create_dataset/main.tf (hunk 1 above)
Comment (JoseFMP) [reply example/create_dataset/main.tf:18]: I would suggest `"google_service_account" "dataset_owner_account"`

File: modules/gcp_bigquery_dataset/outputs.tf [hunk 4]
```diff
@@ -5,3 +5,7 @@ output "id" {
 output "self_link" {
   value = google_bigquery_dataset.google_bigquery_dataset.self_link
 }
+
+output "customer_key_id" {
```
Comment (jai) [inline modules/gcp_bigquery_dataset/outputs.tf:9]: I don't think anybody knows what a 'customer key' is...I think you mean "customer-managed key"? https://cloud.google.com/storage/docs/encryption/customer-managed-keys

**Review by JoseFMP — APPROVED** [review]

File: example/create_dataset/main.tf (hunk 1 above)
Comment (maetolay) [reply example/create_dataset/main.tf]: Fixed to `owner`

File: example/attach_iam_to_dataset/main.tf (hunk 2 above)
Comment (maetolay) [reply example/attach_iam_to_dataset/main.tf]: fixed typo

File: modules/gcp_bigquery_dataset/outputs.tf (hunk 4 above)
Comment (maetolay) [reply modules/gcp_bigquery_dataset/outputs.tf]: Fixed to `customer_managed_key_id`

File: example/attach_iam_to_dataset/main.tf (hunk 3 above)
Comment (maetolay) [reply example/attach_iam_to_dataset/main.tf:27]: @jai I don't understand what you mean?

Comment (maetolay) [reply example/attach_iam_to_dataset/main.tf:30]: @jai add more description

File: modules/gcp_bigquery_dataset_iam_policy/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<!-- BEGIN_TF_DOCS -->
```
Comment (jai) [inline modules/gcp_bigquery_dataset_iam_policy/README.md:1]: If this is a separate module we should at least have a few lines describing what it does

**Review by jai — APPROVED** [review]
please address README for the `bigquery_dataset_iam_policy` thanks

**Pull request merged**